crates/dyn-abi/src/eip712/resolver.rs

```diff
@@ -496,6 +496,24 @@ impl Resolver {
     pub fn contains_type_name(&self, name: &str) -> bool {
         self.nodes.contains_key(name)
     }
+
+    /// Returns the names of those types which do not depend on any other nodes in the resolver
+    /// graph.
+    pub fn non_dependent_types(&self) -> Vec<&str> {
```

Comment:
should we return the TypeDef here as well, so instead of keys() we do iter(), (&str, &TypeDef)